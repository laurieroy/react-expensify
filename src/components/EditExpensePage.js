import React from 'react';

const EditExpensePage = () => (
	<div>
		Editing the expense with id of {props.match.params.id}
	</div>
)

export default EditExpensePage;