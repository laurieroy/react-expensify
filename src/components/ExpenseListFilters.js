import React from "react";
import { DateRangePicker } from "react-dates";
import { connect } from "react-redux";
import {
  setEndDate,
  setStartDate,
  setTextFilter,
  sortByAmount,
  sortByDate,
} from "../actions/filters";

export class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null,
  };

  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };

  onFocusChange = (calendarFocused) => {
    this.setState(() => ({ calendarFocused }));
  };

  onSortChange = (e) => {
    if (e.target.value === "date") {
      this.props.sortByDate();
    } else if (e.target.value === "amount") {
      this.props.sortByAmount();
    }
  };

  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  };

  render() {
    return (
      <div>
        <input
          type="text"
          value={this.props.filters.text}
          onChange={this.onTextChange}
        />

        <select value={this.props.filters.sortBy} onChange={this.onSortChange}>
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker
          startDate={this.props.filters.startDate}
          endDate={this.props.filters.EndDate}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
          showClearDates={true}
          numberOfMonths={1}
          isOutsideRange={() => false}
        />
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByAmount: () => dispatch(sortByAmount()),
  sortByDate: () => dispatch(sortByDate()),
  setEndDate: (endDate) => dispatch(setEndDate(endDate)),
  setStartDate: (startDate) => dispatch(setStartDate(startDate)),
});

const mapStateToProps = (state) => ({ filters: state.filters });

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);
