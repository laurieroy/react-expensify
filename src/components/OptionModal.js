import React from 'react';
import ReactDOM from 'react-dom';
import Modal from 'react-modal';

const OptionModal = (props) => (
	<Modal
		className="modal"
		closeTimeoutMS={200}
		contentLabel="Selected Option"
		isOpen={!!props.selectedOption}
		onRequestClose={props.handleCloseModal}
	>
		<h3 className="modal__title">Selected Option</h3>
		{props.selectedOption && <p className="modal__body">{props.selectedOption}</p>}
		<button className="button" onClick={props.handleCloseModal}>OK</button>
	</Modal>
);

export default OptionModal