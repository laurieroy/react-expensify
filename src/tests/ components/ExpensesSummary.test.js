import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary} from '../../components/ExpensesSummary';

test('should return correct summary for 1 item', () => {
	const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={250} />);
	expect(wrapper).toMatchSnapshot();
});

test('should return correct summary for +1 items', () => {
	const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={750} />);
	expect(wrapper).toMatchSnapshot();
});