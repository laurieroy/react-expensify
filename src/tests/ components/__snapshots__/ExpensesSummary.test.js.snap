// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return correct summary for +1 items 1`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     totalling 
    $7.50
  </h1>
</div>
`;

exports[`should return correct summary for 1 item 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.50
  </h1>
</div>
`;
